sdk/core/Azure.Core.Expressions.DataFactory/src/DataFactoryElementJsonConverter.cs

```diff
@@ -190,6 +196,12 @@ private static void Serialize<T>(Utf8JsonWriter writer, DataFactoryElement<T?> e
                         }
                         writer.WriteEndObject();
                         break;
+                    case BinaryData binaryData:
+                        using (JsonDocument document = JsonDocument.Parse(binaryData.ToString()))
```

Comment:
Nit: do you know that your binary data always contains JSON, and not a base64 encoded string?

Reply:
The service only supports JSON objects. Once ContentType is available on BinaryData we can make this more robust though.